server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java

```diff
@@ -57,7 +63,17 @@
     }
 
     public RemoteConnectionInfo(StreamInput input) throws IOException {
-        seedNodes = input.readList(TransportAddress::new);
+        if (input.getVersion().onOrAfter(Version.V_7_0_0)) {
```

Comment:
Is the bwc aspect of serialization sufficiently tested? I do see `RemoteClusterConnectionTests#testRemoteConnectionInfoBwComp`, just wondering whether that is good enough.

Reply:
I think that's good enough. It shows that if a < 7.0.0 node sends us how it serializes messages today that we can de-serialize it and get what we expect, and shows that if we are sending to a < 7.0.0 that we serialize as that node expects. Also, those versions will change to 6.6.0 when this is backported.